### PR TITLE
Parallelize LibriSpeech data preparation

### PR DESCRIPTION
**Pull request by AsuMagic**
# Contribution in a nutshell

Nyoom:

```
librispeech_prepare - Data_preparation...
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/train-clean-100.csv...
100%|██████████| 28539/28539 [00:01<00:00, 19374.00it/s]
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/train-clean-360.csv...
100%|██████████| 104014/104014 [00:03<00:00, 30984.93it/s]
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/train-other-500.csv...
100%|██████████| 148688/148688 [00:05<00:00, 29710.59it/s]
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/dev-clean.csv...
100%|██████████| 2703/2703 [00:00<00:00, 2764.83it/s]
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/test-clean.csv...
100%|██████████| 2620/2620 [00:00<00:00, 2676.63it/s]
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM/2602/test-other.csv...
100%|██████████| 2939/2939 [00:01<00:00, 2937.20it/s]
speechbrain.dataio.dataio - results/CRDNN_BPE_960h_LM/2602/train.csv is created.
```

Those are the results of data preparation where the data is *local* to the compute node, with 24 threads allocated (but realistically, it doesn't manage to saturate them, because the steps are way fast enough).

Note that the progress bar doesn't include some of the steps involved, most notably crawling the files, which can be _slow_ on NFS. Still, this is a major improvement in my testing.

See commit message for a couple more details.

# Scope
* [x] Parallelize LibriSpeech CSV building
* [x] Use `torchaudio.load` (through SB's `read_audio_metadata`) to greatly speed up the processing step

# Notes for reviewing (optional)

The `diff` between the prepared `train.csv` files between this PR and without the PR is identical on my system.

# Pre-review
* [x] create a fresh testing environment (install SpeechBrain from cloned repo branch of this PR)
* [x] use CI locally: `pre-commit run -a` to check linters; run `pytest tests/consistency`

### CODE REVIEW

File: recipes/LibriSpeech/librispeech_prepare.py [hunk 1]
```diff
@@ -296,30 +325,23 @@ def create_csv(
...
-
-        snt_id = wav_file.split("/")[-1].replace(".flac", "")
-        spk_id = "-".join(snt_id.split("-")[0:2])
-        wrds = text_dict[snt_id]
-
-        signal, fs = torchaudio.load(wav_file)
-        signal = signal.squeeze(0)
-        duration = signal.shape[0] / SAMPLERATE
-
+    for row in parallel_map(line_processor, wav_lst, chunk_size=8192):
```
Comment (Adel-Moumen) [inline recipes/LibriSpeech/librispeech_prepare.py:330]: why you hardcoded the `chunk_size` value while in the CommonVoice recipe this is not the case?

**Review by Adel-Moumen — APPROVED** [review]
Everything is working as expected. I managed to see a huge improvement versus the previous version. Many thanks. (btw I checked the files with `diff` and no issues has been raised). 

librispeech_prepare - Data_preparation...
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-clean-100.csv...
100%|█████████████████████████████████████████████████████████████████████████████████| 28539/28539 [00:13<00:00, 2060.72it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-clean-100.csv successfully created!
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-clean-360.csv...
100%|███████████████████████████████████████████████████████████████████████████████| 104014/104014 [00:47<00:00, 2194.13it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-clean-360.csv successfully created!
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-other-500.csv...
100%|███████████████████████████████████████████████████████████████████████████████| 148688/148688 [01:09<00:00, 2141.41it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/train-other-500.csv successfully created!
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/dev-clean.csv...
100%|████████████████████████████████████████████████████████████████████████████████████| 2703/2703 [00:21<00:00, 123.66it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/dev-clean.csv successfully created!
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/test-clean.csv...
100%|████████████████████████████████████████████████████████████████████████████████████| 2620/2620 [00:03<00:00, 771.44it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/test-clean.csv successfully created!
librispeech_prepare - Creating csv lists in  results/CRDNN_BPE_960h_LM_SLiGRU/2602/test-other.csv...
100%|████████████████████████████████████████████████████████████████████████████████████| 2939/2939 [00:14<00:00, 198.93it/s]
librispeech_prepare - results/CRDNN_BPE_960h_LM_SLiGRU/2602/test-other.csv successfully created!

File: recipes/LibriSpeech/librispeech_prepare.py (hunk 1 above)
Comment (AsuMagic) [reply recipes/LibriSpeech/librispeech_prepare.py:330]: I found out that `torchaudio.info` is already very fast for these FLAC files (to the point parallelism is basically just extra). Bringing the chunk size up decreases overhead and increases worker CPU usage (thus effective utilization) in my testing.

For CommonVoice, I'm pretty sure workers basically already maxxed out all allocated cores, so it wasn't a problem there. We could increase chunk size without many drawbacks other than that the progress bar would update way more slowly but I don't think it'd really help in practice.

Comment (Adel-Moumen) [reply recipes/LibriSpeech/librispeech_prepare.py:330]: Ok ty. Could you please add just a small comment explaining that?

Comment (AsuMagic) [inline recipes/LibriSpeech/librispeech_prepare.py:330]:
```suggestion
    # FLAC metadata reading is already fast, so we set a high chunk size
    # to limit main thread CPU bottlenecks
    for row in parallel_map(line_processor, wav_lst, chunk_size=8192):
```

**Comment by TParcollet** [comment]
Hi both, is the bug that I mentioned with CV data prep solved? Some times it stopped. If here we are using the same strategy for data prep, we could have the same issue. And it would be catastrophic because LS certainly is one of the most used recipe.

**Comment by AsuMagic** [comment]
> Hi both, is the bug that I mentioned with CV data prep solved? Some times it stopped. If here we are using the same strategy for data prep, we could have the same issue. And it would be catastrophic because LS certainly is one of the most used recipe.

#1960 should fix that issue, could you check if you can still repro?

**Comment by Adel-Moumen** [comment]
Should have been fixed in #1960. I did not ran in any issues with the new LS fast prepare.

**Comment by Adel-Moumen** [comment]
Could you please pull the latest dev branch? Ty.

**Comment by AsuMagic** [comment]
Rebased against latest develop.

**Comment by Adel-Moumen** [comment]
Thank you very much @AsuMagic ! :-)

**Pull request merged**